app/components/UI/FiatOnRampAggregator/orderProcessor/customOrderId.ts

```diff
@@ -0,0 +1,74 @@
+import { OrderStatusEnum } from '@consensys/on-ramp-sdk';
+import AppConstants from '../../../../core/AppConstants';
+import { CustomIdData } from '../../../../reducers/fiatOrders/types';
+import { SDK } from '../sdk';
+
+const POLLING_FREQUENCY = AppConstants.FIAT_ORDERS.POLLING_FREQUENCY;
+
+export function createCustomOrderIdData(
+  id: string,
+  chainId: string,
+  account: string,
+): CustomIdData {
+  return {
+    id,
+    chainId,
+    account,
+    createdAt: Date.now(),
+    lastTimeFetched: 0,
+    errorCount: 0,
+  };
+}
+
+export default async function processCustomOrderIdData(
+  customOrderIdData: CustomIdData,
+) {
+  const now = Date.now();
+
+  /**
+   * If the custom order had errors, we don't fetch it unless
+   * POLLING_FRECUENCY ^ (errorCount + 1) seconds has passed
+   */
+  if (
+    customOrderIdData.errorCount > 0 &&
+    customOrderIdData.lastTimeFetched +
+      Math.pow(POLLING_FREQUENCY / 1000, customOrderIdData.errorCount + 1) *
+        1000 >
+      now
+  ) {
+    return [customOrderIdData, null];
+  }
+
+  try {
+    const orders = await SDK.orders();
+    const updatedCustomOrderIdData = await orders.getOrder(
+      customOrderIdData.id,
+      customOrderIdData.account,
+    );
+
+    if (updatedCustomOrderIdData.status === OrderStatusEnum.Unknown) {
+      /** This represents an error, we update the error count and the last time fetched */
+      return [
+        {
+          ...customOrderIdData,
+          lastTimeFetched: Date.now(),
+          errorCount: customOrderIdData.errorCount + 1,
+        },
+        null,
+      ];
+    } else if (updatedCustomOrderIdData.status === OrderStatusEnum.IdExpired) {
+      /** In this case the order is expired and will be removed from the list */
+      return [{ ...customOrderIdData, expired: true }, null];
+    } else if (updatedCustomOrderIdData.status !== OrderStatusEnum.Precreated) {
+      /** In this case the order is not error, expired or precreated, so it is an actual order */
```

Comment:
don't we have an explicit final state for actual order? something like `OrderStatusEnum.valid`

Reply:
An order at this point can be `PENDING`, `COMPLETED`, `CANCELLED` or `FAILED`. It will be added to the orders in any case.

Reply:
In that case I prefer we create a helper function to explicitly check for one of the valid 4 different states: 
```suggestion
      /** In this case the order is not error, expired or precreated, so it is an actual order */
      const isValid = (orderState) => 
           orderState === OrderStatusEnum.PENDING ||
           orderState === OrderStatusEnum.COMPLETED ||
           orderState === OrderStatusEnum.CANCELLED ||
           orderState === OrderStatusEnum.FAILED 
      else if (isValid(updatedCustomOrderIdData.status)) {
           return [customOrderIdData, updatedCustomOrderIdData];
      }
```

Reply:
I can leave a comment saying that given the value of `OrderStatusEnum` these are the only cases possible, but makes no sense to check it programmatically, it is going to be always true.